node/consensus/src/lib.rs

```diff
@@ -265,7 +265,17 @@ impl<N: Network> Consensus<N> {
             if let Err(e) = self.primary_sender().send_unconfirmed_solution(solution_id, Data::Object(solution)).await {
                 // If the BFT is synced, then log the warning.
                 if self.bft.is_synced() {
-                    warn!("Failed to add unconfirmed solution '{}' to the memory pool - {e}", fmt_id(solution_id));
+                    // If error occurs in the first 10 blocks of the epoch, then log it as a trace, otherwise log it as a warning.
+                    match self.ledger().latest_block_height() % N::NUM_BLOCKS_PER_EPOCH <= 10 {
```

Comment:
Why check against the first 10 blocks in the epoch? 

The more comprehensive check is to see if the solution is using the `epoch_hash` of the previous epoch. But that might incur a lookup of some sort or a cache

Reply:
Great point!
The intention of the PR was to reduce warning logs that would occur around epoch changes, hence the first 10 blocks.

I attempted an alternate solution (in commit 9b3f8ca) where we downgrade the log if the solution's epoch hash does not match the existing one. It does not exactly look at the previous epoch hash, but I feel wary about adding caching just for the sake of reduced logs.